shiny/_utils.py

```diff
@@ -362,10 +362,10 @@ def is_async_callable(
     return False
 
 
-# def not_is_async_callable(
-#     obj: Callable[P, T] | Callable[P, Awaitable[T]]
-# ) -> TypeGuard[Callable[P, T]]:
-#     return not is_async_callable(obj)
+def not_is_async_callable(
+    obj: Callable[P, T] | Callable[P, Awaitable[T]],
+) -> TypeGuard[Callable[P, T]]:
+    return not is_async_callable(obj)
```

Comment:
Instead of adding this function, can we change `is_async_callable()` to a `TypeIs`?

Reply:
I just saw the comment below about how `TypeIs` didn't work. Can you add a comment explaining the problem? Otherwise we'll keep coming back to this and asking `TypeIs` in the future.